src/integration/data/deleteMany.test.ts

```diff
@@ -0,0 +1,89 @@
+import { Pinecone, Index } from '../../index';
+import {
+  randomString,
+  createIndexIfDoesNotExist,
+  generateRecords,
+} from '../test-helpers';
+
+describe('deleteMany', () => {
+  const INDEX_NAME = 'ts-integration';
+  let pinecone: Pinecone, index: Index, ns: Index, namespace: string;
+
+  beforeAll(async () => {
+    pinecone = new Pinecone();
+
+    await createIndexIfDoesNotExist(pinecone, INDEX_NAME);
+
+    namespace = randomString(16);
+    index = pinecone.index(INDEX_NAME);
+    ns = index.namespace(namespace);
+  });
+
+  afterAll(async () => {
+    await pinecone.listIndexes();
+    await ns.deleteAll();
+  });
+
+  test('verify deleteMany with ids', async () => {
+    const recordsToUpsert = generateRecords(5, 3);
+    await ns.upsert(recordsToUpsert);
+
+    // Check records got upserted
+    let stats = await ns.describeIndexStats();
+    if (stats.namespaces) {
+      expect(stats.namespaces[namespace].recordCount).toEqual(3);
+    } else {
+      fail('Expected namespaces to be defined');
+    }
+
+    // Look more closely at one of the records to make sure values set
+    const fetchResult = await ns.fetch(['0']);
+    const records = fetchResult.records;
+    if (records) {
+      expect(records['0'].id).toEqual('0');
+      expect(records['0'].values.length).toEqual(5);
+    } else {
+      fail(
+        'Did not find expected records. Fetch result was ' +
+          JSON.stringify(fetchResult)
+      );
+    }
+
+    // Try deleting 2 of 3 vectors
+    await ns.deleteMany(['0', '2']);
+    stats = await ns.describeIndexStats();
+    if (stats.namespaces) {
+      expect(stats.namespaces[namespace].recordCount).toEqual(1);
+    } else {
+      fail(
+        'Expected namespaces to be defined (second call). Stats were ' +
+          JSON.stringify(stats)
+      );
+    }
+
+    // Check that record 1 still exists
```

Comment:
**nit:** Maybe off-by-one in my head, but this should should maybe be "record 2" to reflect the failure response and variable naming here.

Reply:
I made some adjustments to make things more clear. Since the records we upsert are generated with a helper, you can't easily see what the ids are. So I added some expects to confirm the three records have ids 0, 1, and 2 before moving on to deleting things.